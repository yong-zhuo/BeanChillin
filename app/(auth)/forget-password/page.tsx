import ForgetForm from "@/components/user-auth/forget-password/ForgetForm";

export const metadata = {
  title: "Forget Password| BeanChillin",
  description: "Forgot your password? No worries!",
};



export default function ForgetPassword() {
  return <ForgetForm />;
}
