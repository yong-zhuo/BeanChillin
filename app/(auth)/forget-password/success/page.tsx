"use client"

import Button from "@/components/common-ui/button/Button"
import Header from "@/components/common-ui/form/Header"
import { useRouter } from "next/navigation";

export default function ForgetPassword() {
  const router = useRouter();
  const handleClick = () => {
    router.push("/login");
  };
  return (
    <>
      <Header
        heading="Sucess!"
        paragraph="Check your email address for further instructions on how to reset your password."
        logo
      />
        <div className="mx-3 px-5">
        <Button
              handleClick={handleClick}
              text="Back to Login"
              src="/misc/arrow-left.svg"
              alt="arrow-left"
              width={20}
              height={20}
              addClass="gap-2 bg-pri text-white hover:bg-slate-400"
              orientation="left"
            />
        </div>
    </>
  );
}
