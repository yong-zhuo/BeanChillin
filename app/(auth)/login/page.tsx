import LoginForm from "@/components/user-auth/login/LoginForm";

export const metadata = {
  title: "Login | BeanChillin",
  description: "Welcome to BeanChillin!",
};


export default function LoginPage() {
  return <LoginForm />;
}
