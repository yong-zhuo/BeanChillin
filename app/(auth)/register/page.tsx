import SignUpForm from "@/components/user-auth/register/SignUpForm";

export default function Register() {
    return <SignUpForm />
}