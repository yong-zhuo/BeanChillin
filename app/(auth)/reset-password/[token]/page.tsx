"use client";

import FormInput from "@/components/common-ui/form/FormInput";
import Header from "@/components/common-ui/form/Header";
import Button from "@/components/common-ui/button/Button";
import { type reset, resetSchema } from "@/lib/schemas/resetSchema";
import { SubmitHandler, useForm } from "react-hook-form";
import { zodResolver } from "@hookform/resolvers/zod";
import { resetFields } from "@/constants/formFields";
import { fieldState } from "@/types/formFieldsState";
import { useToast } from "@/components/common-ui/shadcn-ui/toast/use-toast";
import { ToastAction } from "@/components/common-ui/shadcn-ui/toast/toast";
import { resetPassword } from "@/lib/mailing/ResetPassword";



//reset password fields to be mapped
const fields = resetFields;
let fieldsState: fieldState = {};
fields.forEach((field) => (fieldsState[field.id] = ""));

const ResetForm = ({ params }: { params: { token: string } }) => {
  //to use shadcn-ui toast
  const { toast } = useToast();

  //zod validation for forget-password
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<reset>({ resolver: zodResolver(resetSchema) });

  //submit handler for forget-password
  const onSubmit: SubmitHandler<reset> = async (data): Promise<void> => {
    //reset password
    try {
      await resetPassword(params.token, data);
    } catch (e: unknown) {
      if (e instanceof Error) {
        toast({
          variant: "destructive",
          title: `Error sending reset link`,
          description: `${e.message}`,
          action: <ToastAction altText="Dismiss">Dismiss</ToastAction>,
        });
      }
    }
  };
  return (
    <>
      <Header
        heading="Reset Password"
        paragraph="Enter your new Password"
        logo
      />
      <form onSubmit={handleSubmit(onSubmit)} className="mb-4 px-40 pb-8 pt-6">
        {fields.map((field) => (
          <FormInput
            key={field.id}
            labelText={field.labelText}
            id={field.id}
            type={field.type}
            placeholder={field.placeholder}
            register={register}
            forRegister
            name={field.name as keyof reset}
            error={errors[field.name as keyof reset]}
          />
        ))}

        <div className="mx-3 px-5">
          <Button
            action="submit"
            text="Reset Password"
            addClass=" text-white bg-pri hover:bg-slate-400"
          />
        </div>
      </form>
    </>
  );
};

export default ResetForm;
