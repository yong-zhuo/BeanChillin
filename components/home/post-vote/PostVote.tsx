"use client";
import { Button } from "@/components/common-ui/shadcn-ui/button";
import usePrevious from "@/hooks/usePrevious";
import { cn } from "@/lib/utils";
import { VoteType } from "@prisma/client";
import { ArrowBigDown, ArrowBigUp } from "lucide-react";
import React, { useContext, useEffect, useState } from "react";
import { UserContext } from "../UserContext";
import { useToast } from "@/components/common-ui/shadcn-ui/toast/use-toast";
import { useRouter } from "next/navigation";

type PostVoteProps = {
  postId: string;
  initVotesCount: number;
  initVote?: VoteType | null;
};

const PostVote = ({
  postId,
  initVotesCount,
  initVote,
}: PostVoteProps) => {
  const [votesCount, setVotesCount] = useState<number>(initVotesCount);
  const [currVote, setCurrVote] = useState(initVote);
  const [isVoting, setIsVoting] = useState(false);
  const prevVote = usePrevious(currVote);
  const {user} = useContext(UserContext);
  const {toast} = useToast();
  const router = useRouter();

  useEffect(() => {
    setCurrVote(initVote);
  }, [initVote]);

  const handleVote = async (type: VoteType) => {

    //prevent multiple votes
    if (isVoting) return;

    setIsVoting(true);

    const payload = {
      voteType: type,
      postId: postId,
      userId:user?.id
    };

    //optimistic update for votes
    if (currVote === type) {
        setCurrVote(undefined);

        if(type === 'UPVOTE') {
            setVotesCount((prevCount) => prevCount - 1);
        } else if (type === 'DOWNVOTE') {
            setVotesCount((prevCount) => prevCount + 1);
        }
    } else {
        setCurrVote(type);

        if(type === 'UPVOTE') {
            setVotesCount((prevCount) => prevCount + (currVote ? 2 : 1));
        } else if (type === 'DOWNVOTE') {
            setVotesCount((prevCount) => prevCount - (currVote ? 2 : 1));
        }
    }

    try {
      const response = await fetch("/api/group/posts/vote", {
        method: "PATCH",
        body: JSON.stringify(payload),
      });

      if (!response.ok) {
        throw new Error("Failed to vote");

      } 

      
        router.refresh();
    

    } catch (error) {
        //revert optimistic update
        if (type === 'UPVOTE') {
            setVotesCount((prevCount) => prevCount - 1);
        } else  {
            setVotesCount((prevCount) => prevCount + 1);
        }
        setCurrVote(prevVote);

        toast({
            title: "Failed to vote",
            description: "Please try again later",
            variant: "destructive",
        
        })
        
    } finally {
        setIsVoting(false);
    }
  };

  return (
    <div className="flex flex-row sm:pl-6 sm:w-12 sm:flex sm:flex-col sm:gap-0 sm:pb-0 justify-center items-center">
      <Button onClick={() => handleVote('UPVOTE')} size="icon" variant="ghost" aria-label="UPVOTE">
        <ArrowBigUp
          className={cn("h-5 w-5 text-zinc-700 hover:bg-sec", {
            "fill-pri text-pri": currVote === "UPVOTE",
          })}
        />
      </Button>
      <p className="m-0 sm:mr-0 py-2 text-center text-sm font-medium text-zinc-900">
        {votesCount}
      </p>
      <Button onClick={() => handleVote('DOWNVOTE')} size="icon" variant="ghost" aria-label="DOWNVOTE">
        <ArrowBigDown
          className={cn("h-5 w-5 text-zinc-700 hover:bg-sec", {
            "fill-red-400 text-red-400": currVote === "DOWNVOTE",
          })}
        />
      </Button>
    </div>
  );
};

export default PostVote;
