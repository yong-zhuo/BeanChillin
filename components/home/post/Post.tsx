"use client";

import UserAvatar from "@/components/common-ui/misc/UserAvatar";
import { formatTimeToNow } from "@/lib/utils";
import type { Post, User, Vote } from "@prisma/client";
import Link from "next/link";

type PartialVote = Pick<Vote, "type">;

interface PostProps {
  post: Post & {
    author: User;
    votes: Vote[];
  };
  votesCount: number;
  groupName: string;
  currVote?: PartialVote;
  commentCount: number;
}

const Post = ({
  post,
  votesCount,
  groupName,
  currVote,
  commentCount,
}: PostProps) => {
  return (
    <div className="rounded-md bg-white shadow">
      <div className="flex justify-between px-6 py-4">
        <div className="w-0 flex-1">
          <div className="mt-1 max-h-40 text-xs text-gray-500">
            <UserAvatar user={post.author} className="h-12 w-12 border-2 border-pri" />
            {groupName ? (
              <>
                <Link
                  className="text-sm text-zinc-900 underline underline-offset-2"
                  href={`/groups/${groupName}`}
                >
                  {groupName}
                </Link>
                <span className="px-1">â€¢</span>
              </>
            ) : null}
            <span>Posted by {post.author.name}</span>{" "}
            {formatTimeToNow(new Date(post.createdAt))}
          </div>
        </div>
      </div>
    </div>
  );
};

export default Post;
