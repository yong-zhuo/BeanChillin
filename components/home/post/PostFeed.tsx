"use client";

import { DetailedPost } from "@/types/post";
import React, { useContext, useEffect, useRef, useState } from "react";
import { UserContext } from "../UserContext";
import { INFINITE_SCROLL_RESULTS } from "@/config";
import { useToast } from "@/components/common-ui/shadcn-ui/toast/use-toast";
import { useInView } from "react-intersection-observer";
import { Loader2 } from "lucide-react";
import Post from "./Post";

interface PostFeedProps {
  initPosts: DetailedPost[];
  groupName?: string;
  feedType?: "general" | "group" | "user" | "trend" | "popular" | "home";
  authorId?: string;
}

const PostFeed = ({ initPosts, groupName, feedType, authorId }: PostFeedProps) => {
  const [offset, setOffset] = useState(INFINITE_SCROLL_RESULTS);
  const [posts, setPosts] = useState<DetailedPost[]>(initPosts);
  const [hasMoreData, setHasMoreData] = useState(true);
  const [scrollTrigger, isInView] = useInView({
    threshold: 1,
  });
  const { user } = useContext(UserContext);
  const { toast } = useToast();

  const loadMorePosts = async () => {
    try {
      if (hasMoreData) {
        const query =
          `/api/posts?limit=${INFINITE_SCROLL_RESULTS}&offset=${offset}` + (!!feedType ? `&feedType=${feedType}` : "") +
          (!!groupName ? `&groupName=${groupName}` : "") + (!!authorId ? `&groupName=${authorId}` : "");
        const res = await fetch(query);
        const newPosts = (await res.json()) as DetailedPost[];

        if (!res.ok) {
          throw new Error("Failed to fetch posts");
        }

        if (newPosts.length === 0) {
          setHasMoreData(false);
        }
        setPosts((prevPosts) => [...prevPosts, ...newPosts]);
        setOffset((prevOffset) => prevOffset + INFINITE_SCROLL_RESULTS);
      }
    } catch (error) {
      toast({
        title: "Failed to fetch posts",
        description: "Please try again later",
        variant: "destructive",
      });
    }
  };

  useEffect(() => {
    if (isInView && hasMoreData) {
      loadMorePosts();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isInView, hasMoreData]);

  return (
    <ul className="col-span-2 flex flex-col space-y-6">
      {posts.map((post, index) => {
        const votesCount = post.votes.reduce((acc, vote) => {
          if (vote.type === "UPVOTE") {
            return acc + 1;
          }
          if (vote.type === "DOWNVOTE") {
            return acc - 1;
          }
          return acc;
        }, 0);

        const currVote = post.votes.find((vote) => vote.userId === user?.id);

        if (index === posts.length - 1) {
          return (
            <li key={post.id} ref={scrollTrigger}>
              <Post
                post={post}
                commentCount={post.comments.length}
                groupName={post.group.name}
                votesCount={votesCount}
                currVote={currVote}
                groupCreatorId={post.group.creatorId as string}
              />
            </li>
          );
        } else {
          return (
            <Post
              key={post.id}
              post={post}
              commentCount={post.comments.length}
              groupName={post.group.name}
              votesCount={votesCount}
              currVote={currVote}
              groupCreatorId={post.group.creatorId as string}
            />
          );
        }
      })}

      {hasMoreData && (
        <li className="flex justify-center" ref={scrollTrigger}>
          Loading
          <Loader2 className="h-6 w-6 animate-spin text-zinc-500" />
        </li>
      )}
    </ul>
  );
};

export default PostFeed;
