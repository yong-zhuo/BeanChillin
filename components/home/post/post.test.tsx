import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { SessionProvider } from 'next-auth/react';
import { Session } from 'next-auth';
import OnboardRoute from '@/app/(onboarding)/onboard/page';
import page from '@/app/(main)/groups/[slug]/post/[postId]/page';
import Page from '@/app/(main)/groups/[slug]/page';
import Post from './Post';
import PostFeed from './PostFeed';
import PostVote from '../post-vote/PostVote';
import { VoteType } from '@prisma/client';
let mockedSession: Session | null = null;


// Mock any necessary modules or hooks here
jest.mock("next-auth/next", () => ({
  getServerSession: jest.fn(() => Promise.resolve(mockedSession)),
}));
jest.mock('next/navigation', () => ({
  useRouter: jest.fn().mockReturnValue({
    push: jest.fn(),
    
  }),
}));


describe('Post', () => {
    it('renders Post correctly and matches snapshot', () => {
        const stubData = {
            id: "post1",
            title: "Example Post Title",
            content: {
                "time": 1719589689482,
                "blocks": [
                  {
                    "id": "i4DGxsAgdC",
                    "data": {
                      "text": "Please share with me where is the best lala soup"
                    },
                    "type": "paragraph"
                  }
                ],
                "version": "2.29.1"
              }, // Assuming Prisma.JsonValue can be an object
            createdAt: new Date("2023-01-01T00:00:00.000Z"),
            updatedAt: new Date("2023-01-02T00:00:00.000Z"),
            authorId: "user1",
            groupId: "group1",
            author: {
              id: "user1",
              name: "John Doe",
              password: "hashed_password",
              signinType: true, // Assuming true or false represents the signinType
              bio: "This is a bio",
              email: "john.doe@example.com",
              emailVerified: new Date("2023-01-01T12:00:00.000Z"),
              imageUrl: "https://example.com/image.jpg",
              imagePublicId: "image1",
              isOnboard: true,
            },
            votes: [
              {
                userId: "user2",
                postId: "post1",
                type: VoteType.DOWNVOTE,
              },
              {
                userId: "user3",
                postId: "post1",
                type: VoteType.DOWNVOTE, 
              },
            ],
          };
      const { asFragment } = render(
        <SessionProvider session={mockedSession}>
          <Post post={stubData} votesCount={1} groupName='1' commentCount={1}/>
        </SessionProvider>
      );
  
      expect(asFragment()).toMatchSnapshot();
      
    });

    it('renders PostVote correctly and matches snapshot', () => {
        
        const { asFragment } = render(
          <SessionProvider session={mockedSession}>
            <PostVote initVotesCount={1} postId='1'/>
          </SessionProvider>
        );
    
        expect(asFragment()).toMatchSnapshot();
        
      });
  });