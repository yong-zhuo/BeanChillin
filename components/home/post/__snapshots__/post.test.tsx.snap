// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders Post correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="rounded-md bg-white shadow"
  >
    <div
      class="flex justify-between px-5 py-4"
    >
      <div
        class="w-0 flex-1"
      >
        <div
          class="mt-1 max-h-40 text-xs text-gray-500"
        >
          <div
            class="flex flex-row"
          >
            <span
              class="relative flex shrink-0 overflow-hidden rounded-full h-12 w-12 border-2 border-pri"
            >
              <div
                class="relative aspect-square h-full w-full"
              >
                <img
                  alt="Profile Picture"
                  data-nimg="fill"
                  decoding="async"
                  fetchpriority="high"
                  referrerpolicy="no-referrer"
                  sizes="100vw"
                  src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=3840&q=75 3840w"
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                />
              </div>
            </span>
            <div
              class="ml-4"
            >
              <div
                class="flex items-center text-lg font-bold text-gray-900"
              >
                <a
                  class="hover:underline"
                  href="/profile/user1"
                >
                  John Doe
                </a>
                <span
                  class="px-1"
                >
                  â€¢
                </span>
                <a
                  class="text-sm text-pri underline underline-offset-2"
                  href="/groups/1"
                >
                  1
                </a>
              </div>
              1y ago
            </div>
          </div>
        </div>
        <a
          href="/groups/1/post/post1"
        >
          <h1
            class="py-2 text-lg font-semibold leading-6 text-gray-900"
          >
            Example Post Title
          </h1>
        </a>
        <div
          class="relative max-h-40 w-full overflow-clip text-sm"
        />
      </div>
      <div
        class="flex items-start justify-center"
      >
        <div
          class="flex flex-row sm:pl-6 sm:w-12 sm:flex sm:flex-col sm:gap-0 sm:pb-0 justify-center items-center"
        >
          <button
            aria-label="UPVOTE"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10"
          >
            <svg
              class="lucide lucide-arrow-big-up h-5 w-5 text-zinc-700 hover:bg-sec"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 18v-6H5l7-7 7 7h-4v6H9z"
              />
            </svg>
          </button>
          <p
            class="m-0 sm:mr-0 py-2 text-center text-sm font-medium text-zinc-900"
          >
            1
          </p>
          <button
            aria-label="DOWNVOTE"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10"
          >
            <svg
              class="lucide lucide-arrow-big-down h-5 w-5 text-zinc-700 hover:bg-sec"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 6v6h4l-7 7-7-7h4V6h6z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="z-20 bg-sec px-4 py-4 text-sm sm:px-6"
    >
      <a
        class="flex w-fit items-center gap-2"
        href="/groups/1/post/post1"
      >
        <svg
          class="lucide lucide-message-square-text h-5 w-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 15a2 2 0 0 1-2 2H7l-4 4V5a2 2 0 0 1 2-2h14a2 2 0 0 1 2 2z"
          />
          <path
            d="M13 8H7"
          />
          <path
            d="M17 12H7"
          />
        </svg>
        1 comments
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Post renders PostVote correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row sm:pl-6 sm:w-12 sm:flex sm:flex-col sm:gap-0 sm:pb-0 justify-center items-center"
  >
    <button
      aria-label="UPVOTE"
      class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10"
    >
      <svg
        class="lucide lucide-arrow-big-up h-5 w-5 text-zinc-700 hover:bg-sec"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 18v-6H5l7-7 7 7h-4v6H9z"
        />
      </svg>
    </button>
    <p
      class="m-0 sm:mr-0 py-2 text-center text-sm font-medium text-zinc-900"
    >
      1
    </p>
    <button
      aria-label="DOWNVOTE"
      class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10"
    >
      <svg
        class="lucide lucide-arrow-big-down h-5 w-5 text-zinc-700 hover:bg-sec"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 6v6h4l-7 7-7-7h4V6h6z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
